doc/GUIDE.md

```diff
@@ -0,0 +1,822 @@
+documentation of the library that you're trying to use to find out which header file should
+be included/passed to jextract.
+
+## Command Line Option Reference
+
+A complete list of all the supported command line options is given below:
+
+| Option                                                       | Meaning                                                      |
+| :----------------------------------------------------------- | ------------------------------------------------------------ |
+| `-D --define-macro <macro>=<value>`                          | define `<macro>` to `<value>` (or 1 if `<value>` omitted)          |
+| `--header-class-name <name>`                                 | name of the generated header class. If this option is not specified, then header class name is derived from the header file name. For example, class "foo_h" for header "foo.h". |
+| `-t, --target-package <package>`                             | target package name for the generated classes. If this option is not specified, then unnamed package is used.  |
+| `-I, --include-dir <dir>`                                    | append directory to the include search paths. Include search paths are searched in order. For example, if `-I foo -I bar` is specified, header files will be searched in "foo" first, then (if nothing is found) in "bar".|
+| `-l, --library <name \| path>`                               | specify a shared library that should be loaded by the generated header class. If <libspec> starts with `:`, then what follows is interpreted as a library path. Otherwise, `<libspec>` denotes a library name. Examples: <br>`-l GL`<br>`-l :libGL.so.1`<br>`-l :/usr/lib/libGL.so.1`|
+| `--use-system-load-library`                                  | libraries specified using `-l` are loaded in the loader symbol lookup (using either `System::loadLibrary`, or `System::load`). Useful if the libraries must be loaded from one of the paths in `java.library.path`.| 
+| `--output <path>`                                            | specify where to place generated files                       |
+| `--dump-includes <String>`                                   | dump included symbols into specified file (see below)        |
+| `--include-[function,constant,struct,union,typedef,var]<String>` | Include a symbol of the given name and kind in the generated bindings. When one of these options is specified, any symbol that is not matched by any specified filters is omitted from the generated bindings. |
+| `--version`                                                  | print version information and exit      
+
+### Additional clang options
+
+Users can also specify additional clang compiler options, by creating a file named
+`compile_flags.txt` in the current folder, as described
+[here](https://clang.llvm.org/docs/JSONCompilationDatabase.html#alternatives).
+
+## Other Languages
+
+As noted in the introduction, jextract currently only supports C header files, but many
```

Comment:
Should we say something about many libraries (e.g. libclang) providing a C interop wrapper around their C++ impl?

Reply:
That's in the table below: `many C++ libraries have a C interface`

Reply:
whoops - true. I thought that row talked about manually writing a C wrapper around a C++ API.